src/main/java/org/broadinstitute/hellbender/tools/walkers/validation/ReferenceBlockConcordance.java

```diff
@@ -0,0 +1,215 @@
+    public static final String TRUTH_BLOCK_HISTOGRAM_LONG_NAME = "truth-block-histogram";
+    public static final String TRUTH_BLOCK_HISTOGRAM_SHORT_NAME = "tbh";
+    public static final String EVAL_BLOCK_HISTOGRAM_LONG_NAME = "eval-block-histogram";
+    public static final String EVAL_BLOCK_HISTOGRAM_SHORT_NAME = "ebh";
+    public static final String CONFIDENCE_CONCORDANCE_HISTOGRAM_LONG_NAME = "confidence-concordance-histogram";
+    public static final String CONFIDENCE_CONCORDANCE_HISTOGRAM_SHORT_NAME = "cch";
+
+    static final String USAGE_ONE_LINE_SUMMARY = "Evaluate GVCF reference block concordance of an input GVCF against a truth GVCF";
+    static final String USAGE_SUMMARY = "This tool evaluates two GVCF files against each other and produces three histograms:\n" +
+            "Truth block histogram: Indicates the number of occurrence of reference blocks with a given confidence score and length in the truth GVCF\n" +
+            "Eval block histogram: Indicates the number of occurrence of reference blocks with a given confidence score and length in the eval GVCF\n" +
+            "Confidence concordance histogram: Reflects the confidence scores of bases in reference blocks in the truth and eval VCF, respectively. An entry of 10 at bin \"80,90\" means that 10 bases in the truth GVCF have a confidence score of 80 while those same bases have a score of 90 in the eval GVCF.\n" +
+            "In contrast to the Concordance tool, this tool considers all variants, regardless of passing or failing filters.";
+
+    @Argument(doc = "A histogram of block lengths and their associated confidence scores for the truth sample",
+            fullName = TRUTH_BLOCK_HISTOGRAM_LONG_NAME,
+            shortName = TRUTH_BLOCK_HISTOGRAM_SHORT_NAME)
+    protected File truthBlockHistogramFile;
+    @Argument(doc = "A histogram of block lengths and their associated confidence scores for the eval sample",
+            fullName = EVAL_BLOCK_HISTOGRAM_LONG_NAME,
+            shortName = EVAL_BLOCK_HISTOGRAM_SHORT_NAME)
+    protected File evalBlockHistogramFile;
+    @Argument(doc = "Reflects the confidence scores of bases in reference blocks in the truth and eval VCF, respectively. An entry of 10 at bin \"80,90\" means that 10 bases in the truth GVCF have a confidence score of 80 while those same bases have a score of 90 in the eval GVCF.",
+            fullName = CONFIDENCE_CONCORDANCE_HISTOGRAM_LONG_NAME,
+            shortName = CONFIDENCE_CONCORDANCE_HISTOGRAM_SHORT_NAME)
+    protected File confidenceConcordanceHistogramFile;
+
+    // TODO this should be a Histogram<Pair<Integer, Integer>>, however, the MetricsFile class cannot read
+    // arbitrary types, therefore, it must be converted to a String, which is probably much slower
+    private final Histogram<String> truthBlockHistogram = new Histogram<>();
+    private final Histogram<String> evalBlockHistogram = new Histogram<>();
```

Comment:
Writing a MetricsFile histogram works with Pair<Integer, Integer>, however, reading that is currently not supported. I could modify htsjdk (specifically FormatUtil) to add support for reading generic objects, or should we just use a string instead?

Reply:
though I could certainly be convinced differently on this

Reply:
I absolutely agree with you that it would be better to use pairs internally, the problem is that Picard writes the type into the header of the metrics file, so it's not possible to write a Histogram<Pair> and read a Histogram<String> unfortunately...